### DISCUSSION TITLE
Always Hidden - How does it work?

### DISCUSSION THREAD

**Discussion by Drallas** [dwarvesf/hidden#162] [Q&A]
I'm not sure if it's and Issue or that I fail to understand the explanation how this works.

After I move an icon left from the grayed out | icon, I right click the > icon; but nothing gets hidden when I collapse the bar icons When I right click > again the moved  icon is on the right of the grayed out | icon again.

What am I missing?

**Comment by viettrungphan** [comment]
Please make sure the position of menu bar buttons look like this:

Apps that you want to hidden | >

If arrow button > at the left hand side of separate bar | the app can't work.

**Reply by Drallas** [reply]
they hide but the always hidden ones don't hide!?

**Reply by viettrungphan** [reply]
Left click to toggle Alway hidden.

**Comment by Drallas** [comment]
Left click doesn't seem to work either and the 'help' (I clicked ? behind 'enable always hidden') shows :
'please right click on the collapse icon to make it disappear' I assume 'collapse icon' is the **<** icon.

So what do I need to right click exactly?

anyway right clicking **<** doesn't toggle Always Hide for me.

**Comment by garfolino** [comment]
It doesn't work for me either.

**Comment by studiohansen** [comment]
i tried out the permanent hiding. it worked. but suddenly i cant see these permanantly hides icons any mor and even not the second devider sign. cant reach these symbols any more - even not after uninstall hidden bar.
help!!!
with every new start of the system these symbols shortly shown up as a menu item, but get lost after a split second.
what can i do - i need them.
a new installation didnt help: now even the other hidden bar symbols are not shown - so i cant do anything...

my system: ventura 13.1 - macbook pro m1 max

**Comment by studiohansen** [comment]
i need immedeatly help. found out, that my mac onl shows max. 6 icons in the menu bar...   showing all starting apps but only max 6 of them...  without having hidden bar startet - permantly.
can i change this somewhere in the system? is there maybe a terminal command?
thank you for any help

**Comment by studiohansen** [comment]
I tried the permanent hiding. It worked. But suddenly I can no longer see these permanent icons and even the second separator. I can't reach these icons anymore - even after uninstalling the hidden bar.
Help!!!
With every restart of the system, these icons are briefly displayed as a menu item, but are lost after a fraction of a second.
What can I do - I need them.
A new installation didn't help: Now not even the other hidden bar icons are displayed - so there's nothing I can do....

**Comment by studiohansen** [comment]
I need help immediately. I found out that my Mac onl shows max 6 icons in the menu bar.... shows all start apps, but only max 6 of them... without starting the hidden bar - permanently.
Can I change this somewhere in the system? Is there perhaps a terminal command?
thanks for any help

**Reply by Drallas** [reply]
"I need help immediately" 🥺☹️
Perhaps ask it [here](https://github.com/dwarvesf/hidden/discussions). 😀

**Comment by Drallas** [comment]
Btw the original issue is resolved for me!

**Comment by zhang-stephen** [comment]
same issue, and how to solve it?

the blurred `|` disappeared after right-clicking the '>' icon.

**Comment by faxotherapy** [comment]
Hi, I still don't get it. I don't understand the difference between “always hidden” and “hidden”. I put a few icons in the “always hidden” section, and the only difference for me it's that those icons are now just icons put on the left side of the little bar. That's all. What's the purpose of this “always hidden” function? Is simple “hidden” not enought? In my case, simply “hidden” just works. Don't see the usefulness of simply “hidden”. Thanks for your help.

**Comment by franciscolourenco** [comment]
I've enabled always-hidden in the preference but still didn't figure out how to make the translucent pipe appear. Tried left/right clicking both the caret and pipe icons. Any tips?

**Comment by mikedvzo** [comment]
I’m experiencing the same issue. The Help menu appears to be confusing, especially because the icons in it are question marks (?) instead of the expected pipe characters (|).

**Comment by dmland** [comment]
The problem may be in the name of the feature _and_ the in-app help, neither of which clarify what the feature even is and how it is expected to behave.

At least on Macos Sequoia, it seems that icons _between_ the "always hidden" and the main "hidden | shown" dividers remain visible after _**right-clicking**_ the ">" chevron icon, while the icons to the left of the "always hidden" divider as well as both dividers are hidden.

On the other hand, all of the icons to the left of the either divider are hidden—along with the dividers—when you **_left-click_** the ">" chevron icon. So the name, "Always hidden" must refer to the fact that icons to the left of the "always hidden" divider are … well … _always hidden_, whether you left-click or right-click, while the icons between the dividers are only hidden on left-click.

So it lets you choose some "medium-importance" icons to keep visible on right-click.

1. Turn on the "always hidden" feature
    - The (half-translucent) "always hidden" divider appears at the far left end of your menu bar.
        - If you have a lot of icons or a smallish screen, it may not be accessible because it's under the menu items of whatever app is on top.
            - Turn on "Use the full MenuBar on expanding" to see more items, but on a small screen, this may not be enough.
            - Change the screen resolution: I use [Display Menu](http://displaymenu.milchimgemuesefach.de/index.html) on my MacBook for a resulution that shows all the icons.
2. Drag it to the right of the icons you want to hide along with the dividers.
    - The position of the other divider becomes somewhat irrelevant in this mode
3. Right-click the ">" chevron icon
    - All of the icons to the left of the "always hidden" divider disappear
    - The "always hidden" and the main "show | hidden" divider both disappear
4. If you left-click the ">" chevron icon, then everything to the left of the main "show | hidden" divider is hidden, but the dividers are not hidden.

**Comment by macmanjimmy** [comment]
Wish someone could post a YouTube video to properly use the "Always Hidden" feature, as it seems a lot are confused by this

**Comment by Damecek** [comment]
Basically when enabled, there will be a special section after extra faded `|` which being toggled by `right` click (show/hidden). 
<img width="383" height="236" alt="image" src="https://github.com/user-attachments/assets/2ed37785-fac0-4ff9-b024-190c390068a6" />
<img width="621" height="81" alt="image" src="https://github.com/user-attachments/assets/79f58197-17d6-4eec-9e3a-f2fe32879789" />

**Reply by do-you-compute** [reply]
The problem for me and apparently others is that right clicking does not hide them. All I have now is a group of unwanted icons that are somewhat spaced farther to the left than the icons I sometimes want, but they all appear/disappear in unison. 

A video would be very helpful.
